library/std/src/io/mod.rs

```diff
@@ -945,6 +945,33 @@ pub trait Read {
     }
 }
 
+/// Convenience function for [`Read::read_to_string`].
+///
+/// This avoids having to create a variable first and it provides more type safety
+/// since you can only get the buffer out if there were no errors. (If you use
+/// [`Read::read_to_string`] you have to remember to check whether the read succeeded
+/// because otherwise your buffer will be empty.)
```

Comment:
This documentation assumes the reader knows the details of `Read::read_to_string`. I think it's better to start with just a description of this function independent of how it relates to `Read::read_to_string` (just like `fs::read_to_string`). Maybe something like:

```rust
/// Read all bytes from a reader into a new [`String`].
///
/// This is a convenience function for using [`Read::read_to_string`] without
/// having to create a variable first. (...)
```